libbeat/publisher/queue/memqueue/queue_test.go

```diff
@@ -175,3 +175,98 @@ func TestAdjustInputQueueSize(t *testing.T) {
 		assert.Equal(t, int(float64(mainQueue)*maxInputQueueSizeRatio), AdjustInputQueueSize(mainQueue, mainQueue))
 	})
 }
+
+func TestEntryIDs(t *testing.T) {
+	entryCount := 100
+
+	testForward := func(q queue.Queue) {
+		producer := q.Producer(queue.ProducerConfig{})
+		for i := 0; i < entryCount; i++ {
+			id, success := producer.Publish(nil)
+			assert.Equal(t, success, true, "Queue publish should succeed")
+			assert.Equal(t, id, queue.EntryID(i), "Entry ID should match publication order")
+		}
+
+		for i := 0; i < entryCount; i++ {
+			batch, err := q.Get(1)
+			assert.NilError(t, err, "Queue read should succeed")
+			assert.Equal(t, batch.Count(), 1, "Returned batch should have 1 entry")
+			assert.Equal(t, batch.ID(0), queue.EntryID(i), "Consumed entry IDs should be ordered the same as when they were produced")
+
+			time.Sleep(100 * time.Millisecond)
```

Comment:
What is this delay for?

Reply:
troubleshooting, it isn't really needed and I'll remove it once I get the last test passing